src/components/Notifications/SubscribeDialog.tsx

```diff
@@ -0,0 +1,140 @@
+"use client";
+
+import { useState } from "react";
+import StarIcon from "./DialogImage/Star";
+import EnvelopeBottom from "./DialogImage/EnvelopeBottom";
+import EnvelopeTop from "./DialogImage/EnvelopeTop";
+import EnvelopePaper from "./DialogImage/EnvelopePaper";
+import { Button } from "../ui/button";
+import { updateNotificationPreferencesForAddress } from "@/app/delegates/actions";
+import { useAccount } from "wagmi";
+import toast from "react-hot-toast";
+import { useDelegate } from "@/hooks/useDelegate";
+
+const HeroImage = ({ isHovering }: { isHovering: boolean }) => {
+  return (
+    <div className="flex items-center gap-2 bg-tertiary/5 rounded-lg relative overflow-y-hidden">
+      <div className="absolute w-full h-full bg-[url('/images/grid.svg')]"></div>
+      <StarIcon
+        className={`text-brandPrimary absolute top-[40px] left-[45px] transition-all duration-500 ${
+          isHovering ? "rotate-[20deg] scale-150" : "rotate-[30deg] scale-125"
+        }`}
+      />
+      <StarIcon
+        className={`text-brandPrimary absolute top-[40px] right-[45px] transition-all duration-500 ${
+          isHovering ? "rotate-[-20deg] scale-150" : "rotate-[-30deg] scale-125"
+        }`}
+      />
+      <StarIcon
+        className={`text-secondary absolute bottom-[40px] left-[55px] transition-all duration-500 ${
+          isHovering ? "rotate-[-20deg] scale-100" : "rotate-[-30deg] scale-95"
+        }`}
+      />
+      <StarIcon
+        className={`text-secondary absolute bottom-[40px] right-[55px] transition-all duration-500 ${
+          isHovering ? "rotate-[20deg] scale-100" : "rotate-[30deg] scale-95"
+        }`}
+      />
+      <div className="mt-2 relative block h-[200px] w-[204px] mx-auto mb-[-50px]">
+        <EnvelopeTop className="absolute bottom-0" />
+        <EnvelopePaper
+          className={`absolute bottom-0 left-4 transition-all ${
+            isHovering ? "bottom-[-8px] duration-500" : "bottom-2 duration-300"
+          }`}
+        />
+        <EnvelopeBottom className="text-brandPrimary absolute bottom-0" />
+      </div>
+    </div>
+  );
+};
+
+const SubscribeDialog = ({
```

Comment:
clicking away from this dialog (not selecting either of the buttons) does nothing. Perhaps we should make it adopt the "decline" behavior? 

If the user clicks away, it will just keep showing them again and again

Reply:
Clicking away with no action seems like a behavior a user would expect. My preference is to leave this as is.

Reply:
Okay thanks, to close this thread then, the behaviors are:

- click away or close modal with the "x" button -> does nothing. The modal will show again.
- click "negative state" button -> interpreted as user intent to decline modal action, do not show modal again.
- click "positive state" button -> interpreted as user intent to confirm modal action, do not show modal again.